src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs

```diff
@@ -1193,20 +1231,36 @@ protected override void EndProcessing()
             {
                 if (ParameterSetName.Equals(InvokeCommandCommand.InProcParameterSet))
                 {
-                    if (_steppablePipeline != null)
+                    Version currentStrictModeVersion = Context.EngineSessionState.CurrentScope.StrictModeVersion;
+                    try
                     {
-                        _steppablePipeline.End();
+                        if (MyInvocation.BoundParameters.ContainsKey(nameof(StrictModeVersion)))
+                        {
+                            Context.EngineSessionState.CurrentScope.StrictModeVersion = _strictmodeversion;
+                        }
+                        
+                        if (_steppablePipeline != null)
+                        {
+                            _steppablePipeline.End();
+                        }
+                        else
+                        {
+                            ScriptBlock.InvokeUsingCmdlet(
+                                contextCmdlet: this,
+                                useLocalScope: !NoNewScope,
+                                errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                                dollarUnder: AutomationNull.Value,
+                                input: _input,
+                                scriptThis: AutomationNull.Value,
+                                args: ArgumentList);
+                        }
                     }
-                    else
+                    finally
                     {
-                        ScriptBlock.InvokeUsingCmdlet(
-                            contextCmdlet: this,
-                            useLocalScope: !NoNewScope,
-                            errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
-                            dollarUnder: AutomationNull.Value,
-                            input: _input,
-                            scriptThis: AutomationNull.Value,
-                            args: ArgumentList);
+                        if (MyInvocation.BoundParameters.ContainsKey(nameof(StrictModeVersion)))
+                        {
+                            Context.EngineSessionState.CurrentScope.StrictModeVersion = currentStrictModeVersion;
```

Comment:
Here we can set the property back to the original setting through the `_savedStrictModeVersion` field.

```csharp
Context.EngineSessionState.CurrentScope.StrictModeVersion = _savedStrictModeVersion;
```